### PR TITLE
use Modal in Hint when onBackgoundPress was passed

### PR DESCRIPTION
**Pull request by ethanshar**

### CODE REVIEW

File: src/components/hint/index.js [hunk 1]
```diff
@@ -132,9 +133,19 @@ class Hint extends BaseComponent {
     }
   };
 
+  get targetLayout() {
```
Comment (Inbal-Tish) [inline src/components/hint/index.js]: This method is called 54 times but the targetLayout only updated twice...

File: src/components/hint/index.js [hunk 2]
```diff
@@ -2,14 +2,15 @@
...
 import {View as AnimatableView} from 'react-native-animatable';
 import {Typography, Spacings, Colors, BorderRadiuses, AnimatableManager} from '../../style';
 import {Constants} from '../../helpers';
 import {BaseComponent} from '../../commons';
 import View from '../view';
 import Text from '../text';
 import Image from '../image';
+import Modal from '../../screensComponents/modal';
 
 const sideTip = require('./assets/hintTipSide.png');
```
Comment (Inbal-Tish) [inline src/components/hint/index.js]: The side tip doesn't render nicely with the borderRadius of the hint...

File: demo/src/screens/componentScreens/HintsScreen.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ export default class HintsScreen extends Component {
       showBottomHint: false,
       showIcon: false,
       targetPosition: 'flex-start',
+      // useTargetFrame: true,
```
Comment (Inbal-Tish) [inline demo/src/screens/componentScreens/HintsScreen.js]: on Android, all the redioButtons are selected in the screen

File: src/components/hint/index.js (hunk 1 above)
Comment (ethanshar) [reply src/components/hint/index.js]: it's just a getter to wrap the logic that returns the relevant `targetLayout`
cause each scenario uses a different value

File: src/components/hint/index.js (hunk 2 above)
Comment (ethanshar) [reply src/components/hint/index.js]: fixed

File: demo/src/screens/componentScreens/HintsScreen.js (hunk 3 above)
Comment (ethanshar) [reply demo/src/screens/componentScreens/HintsScreen.js]: what do u mean?

Comment (Inbal-Tish) [reply demo/src/screens/componentScreens/HintsScreen.js]: When I run it in the device all the buttons are marked blue - selected.

Comment (ethanshar) [reply demo/src/screens/componentScreens/HintsScreen.js]: not sure why, when I run on different emulators it looks good. 
anyway, if anything it's a radioButton issue, let's create a separate ticket to check it.

**Pull request merged**